docs/iris/src/further_topics/metadata.rst

```diff
@@ -0,0 +1,987 @@
+
+    >>> longitude.metadata.difference(metadata)
+    DimCoordMetadata(standard_name=None, long_name=(None, 'lon'), var_name=('longitude', 'lon'), units=(Unit('degrees'), Unit('radians')), attributes=None, coord_system=None, climatological=None, circular=None)
+
+The ``difference`` method returns a :class:`~iris.common.metadata.DimCoordMetadata` instance, when
+there is **at least** one ``metadata`` member with a different value, where,
+
+- ``None`` means that there was **no** difference for the member,
+- a `tuple`_ containing the two different associated values for the member
+
+Given our example, only the ``long_name``, ``var_name`` and ``units`` members
+have different values, as expected. Note that, the ``difference`` method **is
+not** commutative. The order of the tuple member values is the same order
+of the metadata class instances being compared, e.g., changing the
+``difference`` instance order is reflected in the result,
+
+.. doctest:: richer-metadata
+
+    >>> metadata.difference(longitude.metadata)
+    DimCoordMetadata(standard_name=None, long_name=('lon', None), var_name=('lon', 'longitude'), units=(Unit('radians'), Unit('degrees')), attributes=None, coord_system=None, climatological=None, circular=None)
+
+Also, when the ``metadata`` being compared **is identical**, then ``None``
+is simply returned,
+
+.. doctest:: richer-metadata
+
+    >>> metadata.difference(metadata) is None
+    True
+
+It's worth highlighting that for the ``attributes`` `dict`_ member, only
+those keys with **different values** or **missing keys** will be returned by the
+``difference`` method. For example, let's customise the ``attributes`` member of
+the following :class:`~iris.common.metadata.DimCoordMetadata`,
+
+.. doctest:: richer-metadata
+
+    >>> attributes = {"grinning face": "😀", "neutral face": "😐"}
+    >>> longitude.attributes = attributes
+    >>> longitude.metadata
+    DimCoordMetadata(standard_name='longitude', long_name=None, var_name='longitude', units=Unit('degrees'), attributes={'grinning face': '😀', 'neutral face': '😐'}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+Then create another :class:`~iris.common.metadata.DimCoordMetadata` with a different
+``attributes`` `dict`_, namely,
+
+- the ``grinning face`` key has the **same value**,
+- the ``neutral face`` key has a **different value**, and
+- the ``upside-down face`` key is **new**
+
+.. doctest:: richer-metadata
+
+    >>> attributes = {"grinning face": "😀", "neutral face": "😜", "upside-down face": "🙃"}
+    >>> metadata = longitude.metadata._replace(attributes=attributes)
+    >>> metadata
+    DimCoordMetadata(standard_name='longitude', long_name=None, var_name='longitude', units=Unit('degrees'), attributes={'grinning face': '😀', 'neutral face': '😜', 'upside-down face': '🙃'}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+Now, let's compare the two above instances for differences, and see what we get,
```

Comment:
I was being slow...

```suggestion
Now, let's compare the two above instances for differences, and we see the attribute member contains a tuple of the differing information as expected, but each value is a dictionary containing the differing key/values,
```

Reply:
@abooton I wasn't as verbose, as I don't quite think was quite necessary, but I've updated to highlight that the difference is in the `attributes` dictionary. Hope that's a happy compromise...